### PR TITLE
Microverse > Project BookStore > MileStone 4 > FB Category Filter

### PR DESCRIPTION
**Pull request by jurgen1c**
## In this branch: 

- Add filter action
- Add filter reducer
- Add category filter component
- Add filter logic

### CODE REVIEW

File: src/components/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const CategoryFilter = ({ event }) => {
+  const categoryList = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi', 'All'];
+
+  return (
+    <div>
+      <select name="categories" onChange={event}>
+        {categoryList.map(item => <option selected={item === 'Action' ? 'selected' : null} value={item} key={`${item}_option`}>{item}</option>)}
+      </select>
+    </div>
```
Comment (SpaYco) [inline src/components/CategoryFilter.js]: * [ ] `CategoryFilter` merges `All` select option with available categories [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/AllReturnsAllCategories.png)
**_NOTE:_** don't add `All` to the array, instead add it inside `select` before the loop.

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] check below 👇 

####  this one is optional since it is required in the previous milestones and not this one, but it's recommended to do:  

- [ ] `BookList` and `BookForm` components are connected to redux, therefore they should be in the `containers` folder.

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by jurgen1c** [comment]
Last requested changes seem to have made pull request unable to merge, but locally shows no problem and won't show conflict here. Please advise.

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****

 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request closed without merging**